### ISSUE TITLE
ImportError: No module named asl_f4_loader.fw_postprocess

### ISSUE THREAD

**Issue by stieg** [autosportlabs/ASL_F4_bootloader#1]
Trying to get things building on a Jesse Vagrant (for mac development) and am hitting this error

``` python
asl_f4_fw_postprocess -f main.bin -o 0x08020000 -b main.bin -i main.ihex
Traceback (most recent call last):
  File "/usr/local/bin/asl_f4_fw_postprocess", line 9, in <module>
    load_entry_point('asl-f4-loader==0.0.6', 'console_scripts', 'asl_f4_fw_postprocess')()
  File "/usr/lib/python2.7/dist-packages/pkg_resources.py", line 356, in load_entry_point
    return get_distribution(dist).load_entry_point(group, name)
  File "/usr/lib/python2.7/dist-packages/pkg_resources.py", line 2476, in load_entry_point
    return ep.load()
  File "/usr/lib/python2.7/dist-packages/pkg_resources.py", line 2190, in load
    ['__name__'])
```

I can't tell if I'm installing it wrong or if there is an actual bug here.  Followed README and tried to install using both `sudo pip -e <path to project>` and `sudo python setup.py install` from within the project.  But no go :(

**Comment by stieg** [comment]
Turns out this was because pip has issues installing this package.  Seems like something about the naming is wrong that its confusing pip (underscores vs dashes perhaps).

**Comment by ghost** [comment]
Can you give more detail on the installation? I have been trying and failing: No module named asl_f4_loader. When building under the dist there is only an .egg file, but seems like the app is using .py. I have been trying to set this up on Mac El Capitan and has been extremely difficult - thanks for your help and time.

**Comment by stieg** [comment]
The install effectively drop the .egg file in the site-packages directory for your python installation.  On Fedora this is under `/usr/lib/python2.7/site-packages/`.  Then when the application runs (asl_f4_loader in your case) the code invokes `load_entry_point('asl-f4-loader==0.0.6', 'console_scripts', 'asl_f4_loader')()` which finds the egg and executes the resources it needs.  The .py files you are looking for are in the egg file:

```
$ unzip -l /usr/lib/python2.7/site-packages/asl_f4_loader-0.0.6-py2.7.egg
Archive:  /usr/lib/python2.7/site-packages/asl_f4_loader-0.0.6-py2.7.egg
  Length      Date    Time    Name
---------  ---------- -----   ----
        0  10-15-2014 01:43   asl_f4_loader/__init__.py
     5261  06-25-2015 23:39   asl_f4_loader/fw_postprocess.py
     7213  06-25-2015 23:39   asl_f4_loader/fw_update.py
     2485  11-26-2015 12:24   asl_f4_loader/cobs.py
    10816  11-26-2015 12:24   asl_f4_loader/xbvc_py.py
      141  11-26-2015 12:27   asl_f4_loader/__init__.pyc
     5530  11-26-2015 12:27   asl_f4_loader/fw_postprocess.pyc
     7310  11-26-2015 12:27   asl_f4_loader/fw_update.pyc
     1869  11-26-2015 12:27   asl_f4_loader/cobs.pyc
    15148  11-26-2015 12:27   asl_f4_loader/xbvc_py.pyc
      538  11-26-2015 12:27   EGG-INFO/PKG-INFO
      349  11-26-2015 12:27   EGG-INFO/SOURCES.txt
        1  11-26-2015 12:27   EGG-INFO/dependency_links.txt
      122  11-26-2015 12:27   EGG-INFO/entry_points.txt
       35  11-26-2015 12:27   EGG-INFO/requires.txt
       14  11-26-2015 12:27   EGG-INFO/top_level.txt
        1  11-26-2015 12:27   EGG-INFO/zip-safe
---------                     -------
    56833                     17 files
```

Hope that helps

**Comment by ghost** [comment]
Finally moved passed, and then ran into issues with pygame, fixed that and now: "pygame - ImportError: cannot import name encode_string" and cannot get the window going. Seems to be issue around 64 when can only support 32 on Mac and possibly Yosemite / newer. I am going to move to Linux Ubuntu 15 to try and get the environment setup. I added an issue for docker image, but then closed because just did not make sense. I might try and decompile the libraries from the mac install and then get the source to develop. I really need to add OBD2 support for Lua and having a complex time just to get the environment for testing working. Thanks for the time and really excited about the possibilities.

**Comment by brentpicasso** [comment]
Hi,

I just installed and ran the dev environment on a Raspberry Pi, of all things. I ran a hybrid of the Kivy / Rpi instructions and our instructions. Check it out here. http://kivy.org/docs/installation/installation-rpi.html

What was interesting was the direct pip install from our repo. example:
sudo pip install git+https://github.com/autosportlabs/kivy@asl_1.9.0_patched 

However, the Linux installation should work starting from here; make sure you get the virtual environment set up as well.
https://github.com/autosportlabs/RaceCapture_App#dev-installation-linux---ubuntu

Let us know exactly what you get hung up, if you do,  on and we can go from there.
-Brent
